src/Components/MailtoDropdown.tsx

```diff
@@ -0,0 +1,171 @@
+import { useModal } from './Modal';
+import t, { t_r } from '../Utility/i18n';
+
+// We only want to select everything in the copymanually inputs if they aren't yet focused. That way, the user can still
+// make an individual selection if they prefer.
+// However, the event we get in the onclick handler means that the focus has already been changed to the element the
+// user clicked, so we need to remember the previous one for the check to make any sense at all.
+let previous_active_element = document.body;
+
+type EmailData = { email: string; subject: string; body: string };
+type MailtoHandler = (
+    | { link: (data: EmailData) => string }
+    | { onClick: (data: EmailData, showCopyManuallyModal: () => void) => void }
+) & { countries: typeof window.globals.country[] };
+
+type MailtoDropdownProps = {
+    // TODO: @zner0L is working on that right now.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    letter?: any;
+    handlers?: (keyof typeof mailto_handlers)[];
+    email: string;
+    onSuccess: () => void;
+    done: boolean;
+    className: string;
+    enabled: boolean;
+    buttonText: JSX.Element | JSX.Element[];
+};
+
+// TS Helper to type `Record` values but keep strong key type without having to hardcode key values, see:
+// https://stackoverflow.com/a/49539369
+const createMailtoHandlers = <T extends Record<string, MailtoHandler>>(handlers: T) => handlers;
+const mailto_handlers = createMailtoHandlers({
+    mailto: {
+        link: (d) => `mailto:${d.email}?subject=${d.subject}&body=${d.body}`,
+        countries: ['all'],
+    },
+    gmail: {
+        link: (d) => `https://mail.google.com/mail/?view=cm&fs=1&to=${d.email}&su=${d.subject}&body=${d.body}`,
+        countries: ['all'],
+    },
+    yahoo: {
+        link: (d) => `https://compose.mail.yahoo.com/?to=${d.email}&subject=${d.subject}&body=${d.body}`,
+        countries: ['all'],
+    },
+    yandex: {
+        link: (d) => `https://mail.yandex.com/#compose?to=${d.email}&subject=${d.subject}&body=${d.body}`,
+        countries: ['all'],
+    },
+    copymanually: {
+        onClick: (d, showCopyManuallyModal) => showCopyManuallyModal(),
+        countries: ['all'],
+    },
+});
+
+export const MailtoDropdown = (props: MailtoDropdownProps) => {
+    const my_ref_text = `${t_r('my-reference', props.letter.props.language)}: ${props.letter.props.reference}`;
+    const data = {
+        email: props.email,
+        subject: encodeURIComponent(
+            props.letter.props.subject ? `${props.letter.props.subject} (${my_ref_text})` : my_ref_text
+        ),
+        body: encodeURIComponent(props.letter.toEmailString()),
+    };
+
+    const onCopyManuallyInputClick = (e: JSX.TargetedMouseEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+        if (previous_active_element.id === e.currentTarget.id) return;
+
+        e.currentTarget.select();
+        e.currentTarget.focus();
+        previous_active_element = e.currentTarget;
```

Comment:
Why is this not a state?

Reply:
I remember that I had a reason for that but I don't remember what it was. :( You could watch the video of me implementing the MailtoDropdown to find out. :P

Reply:
You didn't have a good reason. You can even just move the id into the state instead of the whole element.